packages/context/SearchBarProvider.tsx

```diff
@@ -0,0 +1,99 @@
+import { useQuery } from "@tanstack/react-query";
+import React, { createContext, useContext, useState } from "react";
+import { useSelector } from "react-redux";
+
+import { Collection } from "../api/marketplace/v1/marketplace";
+import { useSelectedNetworkId } from "../hooks/useSelectedNetwork";
+import { selectSearchText } from "../store/slices/search";
+import { getMarketplaceClient } from "../utils/backend";
+
+interface DefaultValue {
+  isSearchModalMobileOpen: boolean;
+  setSearchModalMobileOpen: (value: boolean) => void;
+  names?: string[];
+  collections?: Collection[];
+  hasCollections?: boolean;
+  hasNames?: boolean;
```

Comment:
```suggestion
  names: string[];
  collections: Collection[];
  hasCollections: boolean;
  hasNames: boolean;
```

we can provide sane default values for these so I recommend to not allow them to be undefined

Reply:
Fixed